Sources/Public/CalendarViewContent.swift

```diff
@@ -407,22 +392,20 @@ public final class CalendarViewContent {
 
   /// The default `monthHeaderItemProvider` if no provider has been configured,
   /// or if the existing provider returns nil.
-  private lazy var defaultMonthHeaderItemProvider: (Month) -> AnyCalendarItemModel = { [
-    calendar,
-    monthHeaderDateFormatter
-  ] month in
-    let firstDateInMonth = calendar.firstDate(of: month)
-    let monthText = monthHeaderDateFormatter.string(from: firstDateInMonth)
-    let itemModel = MonthHeaderView.calendarItemModel(
-      invariantViewProperties: .base,
-      content: .init(monthText: monthText, accessibilityLabel: monthText))
-    return itemModel
-  }
+  private lazy var defaultMonthHeaderItemProvider: (Month) -> AnyCalendarItemModel = 
+    { [unowned self] month in
```

Comment:
Should be impossible to hit cc @calda

Reply:
What do you mean?

Reply:
@calda I mean the unowned should be "safe" / there's no way to crash here.

Reply:
Perhaps -- although not sure I understand the motivation to change from capturing individual properties (guaranteed to be safe) to using `unowned` instead (potentially unsafe).

You could feasibly construct a case that crashes by doing something like this:

```swift
let content = CalendarViewContent(...)

let monthHeaderItemProvider = content.monthHeaderItemProvider

/// ... later, in a different scope, so content itself has been deallocated

monthBackgroundItemProvider(month) // 💥 crash because content no longer exists but there was an unowned reference to it
```

so the safety depends on the usage of the `content.monthHeaderItemProvider` property

Reply:
That would only be possible internally though, since CalendarViewContent doesn't expose its item provider closures publicly.

Capturing the individual properties here causes them to be created immediately (which is expensive for date formatters).

```swift
lazy var expensiveProperty = ExpensiveObject()

let foo = { [expensiveProperty] in 
  expensiveProperty.doSomething()
}

// later...
foo()
```

In this example, `expensiveProperty` is initialized when `foo` is initialized, _not_ when `foo` is called later on.

Reply:
That makes sense -- I guess in a lot of typical usage we'd be initializing the `monthHeaderDateFormatter` for the `defaultMonthHeaderItemProvider` but then never even calling the default provider (because the consumer immediately provides a custom provider)